File: 365-water-and-jug-problem/water-and-jug-problem.cpp
class Solution {
public:
    int GCD(int a, int b){
        while (b>0){
            int c = a%b;
            a = b;
            b=c;
        }
        return a;
    }

    bool canMeasureWater(int x, int y, int target) {
        if(target > (x+y)){
            return false;
        }

        if(x == target || y == target || target == (x+y) || target % GCD(x,y) == 0){
            return true;
        } else {
            return false;
        }
    }
};

File: 365-water-and-jug-problem/water-and-jug-problem.py
class Solution(object):

    def canMeasureWater(self, x, y, target):
        """
        :type x: int
        :type y: int
        :type target: int
        :rtype: bool
        """
        def GCD(a,b):
            while b>0:
                [a,b] = [b,a%b]
            return a

        if(target > x+y):
            return False
        elif (x == target or y == target or target == x+y):
            return True
        elif (target % GCD(x,y) == 0):
            return True

        return False
        